### PR TITLE
Update Multiplayer Tests to Store their Cached Folder Level Assets

### PR DESCRIPTION
**Pull request by AMZN-Gene**
Update Multiplayer tests to store their cached level assets (which contains the spawnable player and scripts) inside the test artifacts. This way we can compare assets if any tests become flaky

Level cache is around 140KB per test, but gets compressed and stored around 30KB
Tested by running AutomatedTesting for multiplayer and seeing the newly add level folders inside the TestResults artifacts.

Signed-off-by: Gene Walters <genewalt@amazon.com>

### CODE REVIEW

**Review by evanchia-ly-sdets — APPROVED** [review]

File: AutomatedTesting/Gem/PythonTests/Multiplayer/TestSuite_Main.py [hunk 1]
```diff
@@ -20,12 +20,36 @@ class TestAutomation(EditorTestSuite):
     class test_Multiplayer_AutoComponent_NetworkInput(EditorSingleTest):
         from .tests import Multiplayer_AutoComponent_NetworkInput as test_module
 
+        @classmethod
+        def setup(cls, instance, request, workspace, editor, editor_test_results, launcher_platform):
+            level_cache_folder_path = os.path.join(workspace.paths.platform_cache(), "levels", "multiplayer", "autocomponent_networkinput")
+            if os.path.exists(level_cache_folder_path):
```
Comment (evanchia-ly-sdets) [inline AutomatedTesting/Gem/PythonTests/Multiplayer/TestSuite_Main.py:26]: I would maybe just suggest logging a debug/warning message if the folder is not found for whatever reason

Comment (AMZN-Gene) [reply AutomatedTesting/Gem/PythonTests/Multiplayer/TestSuite_Main.py]: Updated! Calling _pytest.fail_ if the asset cache wasn't found.

File: AutomatedTesting/Gem/PythonTests/Multiplayer/TestSuite_Main.py [hunk 2]
```diff
@@ -13,19 +13,43 @@
 
 sys.path.append(os.path.dirname(os.path.abspath(__file__)) + '/../automatedtesting_shared')
 
+def save_multiplayer_level_cache_folder_artifact(workspace, multiplayer_level):
```
Comment (lmbr-pip) [inline AutomatedTesting/Gem/PythonTests/Multiplayer/TestSuite_Main.py:16]: nit: consider comments on public methods and typing hints

File: AutomatedTesting/Gem/PythonTests/Multiplayer/TestSuite_Main.py [hunk 3]
```diff
@@ -13,19 +13,43 @@
 
 sys.path.append(os.path.dirname(os.path.abspath(__file__)) + '/../automatedtesting_shared')
```
Comment (lmbr-pip) [inline AutomatedTesting/Gem/PythonTests/Multiplayer/TestSuite_Main.py:14]: nit: add a comment to why you are appending to the system path.

**Review by kberg-amzn — APPROVED** [review]

File: AutomatedTesting/Gem/PythonTests/Multiplayer/TestSuite_Main.py (hunk 2 above)
Comment (AMZN-Gene) [reply AutomatedTesting/Gem/PythonTests/Multiplayer/TestSuite_Main.py:17]: thank you! fixed!

File: AutomatedTesting/Gem/PythonTests/Multiplayer/TestSuite_Main.py (hunk 3 above)
Comment (AMZN-Gene) [reply AutomatedTesting/Gem/PythonTests/Multiplayer/TestSuite_Main.py]: removed! along with some imports i didn't need.

**Pull request merged**